File: 435-Non-overlapping Intervals.py
class Solution(object):
    def eraseOverlapIntervals(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: int
        """
        
        intervals.sort(key = lambda x:x[1])
        count = 0
        
        prev = float('-inf')
        for interval in intervals:
            if interval[0] >= prev:
                prev = interval[1]
            else:
                count += 1
        return count

File: JavaScript/435-Non-overlapping Intervals.js
/**
 * @param {number[][]} intervals
 * @return {number}
 */
var eraseOverlapIntervals = function(intervals) {
    intervals.sort(function(a,b){return a[1]-b[1]});
    
    let prev = Number.NEGATIVE_INFINITY, count=0;
    for (let interval of intervals){
        if (interval[0] >= prev){
            prev = interval[1];
        }else{
            count += 1;
        }
    }
    return count;
};